### PR TITLE
Docs setup

### PR DESCRIPTION
**Pull request by loferris**
Adds current SeqIO documentation as a single page from the index.

### CODE REVIEW

File: seqio/docs/source/overview.md [hunk 1]
```diff
@@ -0,0 +1,962 @@
...
+registry, if desired.
+
+We'll now break down each part of the task definition.
+
+#### Data Source
+
+Data sources are the first step in your pipeline, providing a way to load raw
+data in many formats as a `tf.data.Dataset`.
+All data sources are subclasses of the `DataSource` base class and are defined in
+[dataset_providers](http://google3/third_party/py/seqio/dataset_providers.py),
```
Comment (gauravmishra) [inline seqio/docs/source/overview.md:103]: change to http://github.com/google/seqio/blob/main/seqio/dataset_providers.py

File: seqio/docs/source/overview.md [hunk 2]
```diff
@@ -0,0 +1,962 @@
...
+The `output_features` field expects a dictionary that maps string feature names
+to `seqio.Feature` objects. This defines what the `Task` is expected to produce
+in its output examples. The output examples *may* contain additional fields, but
+they *must* contain these fields in the specified format or exceptions will be
+raised.
+
+
+Each `Feature` includes:
+
+  * A `vocabulary`, which must subclass [`seqio.Vocabulary`](http://google3/third_party/py/seqio/vocabularies.py), to specify how the feature can be tokenized and detokenized. You may use `seqio.PassThroughVocabulary` if tokenization is not necessary.
```
Comment (gauravmishra) [inline seqio/docs/source/overview.md:125]: change to https://github.com/google/seqio/blob/main/seqio/vocabularies.py

File: seqio/docs/source/overview.md [hunk 3]
```diff
@@ -0,0 +1,962 @@
...
+        else:
+          new_ex[k] = v
+      return new_ex
+
+  return take_chunk(dataset)
+  ```
+
+  If `num_seeds > 1`, the arg will instead be called `seeds` and will contain a sequence of seeds.
+
+In our "wmt_19_ende" task, we also use the predefined preprocessors `seqio.preprocessors.tokenize` and `seqio.preprocessors.append_eos`. The former uses each `Feature.vocabulary` to tokenize it, and the the latter appends `Feature.vocabulary.eos_id` to the feature if the `Feaure.add_eos` is True. See [preprocessors.py](http://google3/third_party/py/seqio/preprocessors.py) for their implementations and other useful preprocessors.
```
Comment (gauravmishra) [inline seqio/docs/source/overview.md:251]: change to https://github.com/google/seqio/blob/main/seqio/preprocessors.py

File: seqio/docs/source/overview.md [hunk 4]
```diff
@@ -0,0 +1,962 @@
...
+Since the postprocess function is used on both the model output and the targets, it is passed an `is_target` boolean in case the behavior should be different. It is also passed the fully preprocessed example, including fields that were excluded from `output_features`.
+
+For the "wmt19_ende", we don't need any postprocessors. See "trivia_qa_open"
+task in the [Advanced Postprocessing `Task`](#advanced-postprocessing-task) for an example postprocessor.
+
+#### Metrics
+
+Metrics are functions that are passed (by the [Evaluator](#evaluator)) the fully-materialized list of postprocessed model outputs (or scores) and targets and return a mapping from string names to `MetricValue` objects containing their values. These are most commonly floating-point scalars, but may also be text, images, audio, histograms, etc (see [metrics.py](http://google3/third_party/py/seqio/metrics.py) for the full list).
```
Comment (gauravmishra) [inline seqio/docs/source/overview.md:263]: change to https://github.com/google/seqio/blob/main/seqio/metrics.py

File: seqio/docs/source/overview.md [hunk 5]
```diff
@@ -0,0 +1,962 @@
...
+  ```py
+  seqio.MixtureRegistry.add(
+    "mix1",
+    [("task1", 0.5), "task2"],
+    default_rate=3.5
+  )
+  ```
+
+  1. Provide a function that generates the rate for each task at runtime.
+  The example below uses the provided [`seqio.mixing_rate_num_examples`](http://google3/third_party/py/seqio/utils.py), which uses the number of examples (computed during [offline caching](#optional-offline-caching)) as the rate for each task.
```
Comment (gauravmishra) [inline seqio/docs/source/overview.md:349]: change to https://github.com/google/seqio/blob/main/seqio/utils.py

File: seqio/docs/source/overview.md [hunk 6]
```diff
@@ -0,0 +1,962 @@
...
+### (Optional) Offline Caching
+
+For improved performance at load time and avoid redundant computations for commonly used tasks, you can pre-cache your `Task` with all or part of the preprocessing done in advance of training.
+
+The first step to doing so is to add a `seqio.CacheDatasetPlaceholder(required=False)` as one of the steps in your preprocessing pipeline. All steps before the placeholder will be cached offline and all steps after will be executed on the fly at load time. You may set `required=True` if you want `get_dataset` to fail unless `use_cached=True`.
+
+Caveats:
+
+* Any stochastic operations that you wish to be re-run when `num_epochs > 1` or with a different `seed` *should* go after the placeholder since only a single sample will be cached.
+* Any preprocessing steps that use the `sequence_length` argument *must* come after the `seqio.CacheDatasetPlaceholder` preproessor since this is only known at runtime, or an exception will be raised. If you wish to cache for a specific sequence length, you can use [`seqio.experimental.add_fully_cached_task`](http://google3/third_party/py/seqio/experimental.py).
```
Comment (gauravmishra) [inline seqio/docs/source/overview.md:410]: change to https://github.com/google/seqio/blob/main/seqio/experimental.py

File: seqio/docs/source/overview.md [hunk 7]
```diff
@@ -0,0 +1,962 @@
...
+Let's use machine translation (English to German) as a running example.
+
+The raw data consists of sentence pairs such as
+
+```
+"That is good\tDas ist gut."
+```
+
+A task registered to `Task` (e.g.,
+[wmt_t2t_ende_v003](http://google3/third_party/py/t5/data/tasks.py?l=156&rcl=337594707))
```
Comment (gauravmishra) [inline seqio/docs/source/overview.md:434]: change to https://github.com/google-research/text-to-text-transfer-transformer/blob/main/t5/data/tasks.py#L212

File: seqio/docs/source/overview.md [hunk 8]
```diff
@@ -0,0 +1,962 @@
...
+reads these sentence pairs from the data source and applies a series of
+[preprocessors](http://google3/third_party/py/t5/data/preprocessors.py?rcl=343354647).
```
Comment (gauravmishra) [inline seqio/docs/source/overview.md:436]: change to https://github.com/google-research/text-to-text-transfer-transformer/blob/main/t5/data/preprocessors.py

**Review by gauravmishra — CHANGES_REQUESTED** [review]
Looks good! A few changes required to scrub internal links.

**Review by gauravmishra — APPROVED** [review]
Looks great, thanks a lot!
